### DISCUSSION TITLE
Using ploidy option versus number of individuals * ploidy

### DISCUSSION THREAD

**Discussion by isinaltinkaya** [tskit-dev/msprime#2046] [Q&A]
Hi,

I have run bunch of simulations aiming to simulate diploid individuals using `number_of_individuals * 2` as sample size. I just noticed that there is also a `ploidy` option. Are there any difference between using `ploidy=2` and `ploidy=default; sample_size=size*2` to simulate diploid samples? I would assume not since these would be haplotype counts, but I could not be sure if this would affect the calculation of any other parameters like recombination rate rho or theta.

Thanks!

**Comment by benjeffery** [comment]
There's more detail about ploidy here if you missed it: https://tskit.dev/msprime/docs/stable/ancestry.html#sec-ancestry-ploidy 
In addition to storing the `individuals` correctly in the tree sequence this argument also affects the time scale on which coalescence occurs.
If you have further questions please ask, I'm converting this issue to a discussion.

**Comment by isinaltinkaya** [comment]
Thank you very much!

I actually wanted to understand the expected behavior compared to ms/msms. I have been using msms before I moved on to using msprime. I have an equivalent demographic model defined for both msms and msprime, and with msms I was using `nsamples*2` to simulate diploid samples. Would this be equivalent to using `nsamples*2` in msprime or `ploidy=2`?

**Reply by jeromekelleher** [reply]
Yes @isinaltinkaya, it is equivalent. We're just arbitrarily pairing up the samples.

However, please note that it does also change the timescaling, so if ploidy!=2 you won't get the same time scales as ms. The docs linked above will hopefully explain this point.

**Reply by isinaltinkaya** [reply]
Thank you @jeromekelleher.

Are the haplotypes ordered based on the relationships or do they represent random samplings?

I want to simulate 10 diploid individuals from 3 populations each `[10,10,10]` therefore 30 individuals in total.

For this, I am using `ploidy=2` and `samples=60` following `[20, 20, 20]`.

I get the following table using `ts.__str__()`:

```
╔═══════════════════════════╗
║TreeSequence               ║
╠═══════════════╤═══════════╣
║Trees          │     119629║
╟───────────────┼───────────╢
║Sequence Length│ 51229805.0║
╟───────────────┼───────────╢
║Time Units     │generations║
╟───────────────┼───────────╢
║Sample Nodes   │         60║
╟───────────────┼───────────╢
║Total Size     │   26.5 MiB║
╚═══════════════╧═══════════╝
╔═══════════╤══════╤═════════╤════════════╗
║Table      │Rows  │Size     │Has Metadata║
╠═══════════╪══════╪═════════╪════════════╣
║Edges      │428617│ 13.1 MiB│          No║
╟───────────┼──────┼─────────┼────────────╢
║Individuals│     0│ 24 Bytes│          No║
╟───────────┼──────┼─────────┼────────────╢
║Migrations │     0│  8 Bytes│          No║
╟───────────┼──────┼─────────┼────────────╢
║Mutations  │113781│  4.0 MiB│          No║
╟───────────┼──────┼─────────┼────────────╢
║Nodes      │ 70344│  1.9 MiB│          No║
╟───────────┼──────┼─────────┼────────────╢
║Populations│     3│131 Bytes│         Yes║
╟───────────┼──────┼─────────┼────────────╢
║Provenances│     1│  1.5 MiB│          No║
╟───────────┼──────┼─────────┼────────────╢
║Sites      │113781│  2.7 MiB│          No║
╚═══════════╧══════╧═════════╧════════════╝
```


I need to print the ms style haplotypes for each individual. I use `tskit.write_ms()` to convert this into ms format:

```
ms 60 1                                                                                                                                       
0

//
segsites: 138643
0100...
0110...
0100...
0101...
```
I want to use these haplotypes to obtain diploid haplotypes. Therefore I am assigning them to individuals in order:

```
0100... -> ind1
0110... -> ind1
0100... -> ind2
0101... -> ind2
```

I would like to learn if this is the proper way to obtain diploid haplotypes, and whether the ordering of these samples is random within populations or e.g. if the first sequence is probably more closely related to the next sequence.

**Reply by jeromekelleher** [reply]
The ``samples`` argument refers to **individuals** now not nodes, so if you want  want to simulate 10 diploid individuals from 3 populations you want ``ts = msprime.sim_ancestry(samples=[10, 10, 10], ploidy=2)``

You can see how the nodes and individuals relate to each other if you print out the node and invidual tables as shown [here](https://tskit.dev/msprime/docs/stable/ancestry.html#default-number-of-nodes-per-sample-individual)

But yes, the samples are paired up and assigned to individuals in order like you show above. There is no particular meaning to how the samples are grouped together though in this model, it's arbitrary.

**Reply by isinaltinkaya** [reply]
I am a bit confused here. When I use `[20,20,20]`, with `write_ms` I get 60 rows in binary haplotypes part of ms output. If I use `[10,10,10]` this will be 30, but I would like to get 2 haplotypes per individual to represent diploid individuals.

**Reply by jeromekelleher** [reply]
Can you paste in your code please @isinaltinkaya, I think I'm probably misunderstanding things here.

**Reply by isinaltinkaya** [reply]
I use msprime to simulate `samples=[20,20,20]` and `ploidy=2`. Then I convert ts file ms format using `write_ms`. 

When I check the number of haplotypes in the ms output, there are 60 haplotypes:
```
$ tail -n+7 test.ms |wc -l
60
```
But since my individuals are diploids, I expected to get 120 lines here (2 lines per individual).

**Reply by jeromekelleher** [reply]
Can you put in the full Python code called to sim_ancestry and write_ms please?

**Reply by isinaltinkaya** [reply]
I just found out that I had the same confusion as described in https://github.com/tskit-dev/msprime/discussions/1761; therefore it is resolved. Thanks a lot!
